modules/cloudfront_distribution/cloudfront_distribution.py

```diff
@@ -39,6 +39,25 @@ def process(self, module_idx: int) -> None:
                 "s3_log_bucket_name", f"${{{{{module_source}.s3_log_bucket_name}}}}"
             )
 
+        aws_dns_modules = self.layer.get_module_by_type("aws-dns", module_idx)
+        if len(aws_dns_modules) != 0 and aws_dns_modules[0].data.get("linked_module") in [
+            self.module.type,
+            self.module.name,
```

Comment:
Should there be a restriction on the what linked_modules can accept?
Or rather some sort of warning or error log, otherwise the user might just add anything, and not know whether it is supposed to work or not.

Reply:
@nsarupr added